docs/semgrep-ci/running-semgrep-ci-with-semgrep-app.md

```diff
@@ -328,11 +329,107 @@ SEMGREP_PR_ID="44"
+- **Repository access token**: If you are **not** using the Bitbucket Cloud Premium plan, you have to create a separate repository access token for each repository where you want to use Semgrep. See [Creating a repository access token](/semgrep-ci/running-semgrep-ci-with-semgrep-app/#creating-and-adding-a-repository-access-token).
+
+###### Creating and adding a workspace access token
+
+:::info Prerequisite
+Use the procedure described in this section if you use the **Bitbucket Cloud Premium** plan. If you are **not** using the Bitbucket Cloud Premium plan, you have to create a separate repository access token for each repository where you want to use Semgrep. See [Creating a repository access token](/semgrep-ci/running-semgrep-ci-with-semgrep-app/#creating-and-adding-a-repository-access-token).
+:::
+
+Create a workspace access token in Bitbucket (only available if you have a Bitbucket Cloud Premium plan). Follow the instructions in [Create a Workspace Access Token](https://support.atlassian.com/bitbucket-cloud/docs/create-a-workspace-access-token/). Alternatively, follow these steps to create a workspace access token:
+
+1. In the Bitbucket workspace, click the cogwheel icon next to your profile icon in the top right.
+1. Click **Workspace settings** > **Access tokens**.
+1. Click **Create Workspace Access Token**.
+1. Create a token name. Note: As Semgrep comments to your PRs, use **Semgrep** as the name for this token.
+1. Under **Scopes**, select the **Read**, and **Write** permissions for pull requests.
+1. Click **Create**.
+1. Copy the value of the workspace access token, and then click **Close**.
+1. Click **Workspace variables**.
+1. Create a new workspace variable by entering **PAT** into the **Name** field, and then paste the workspace access token in the **Value** field.
+1. Enable the **Secured** option, and then click **Add**.
+
+Create and add a `SEMGREP_APP_TOKEN` to establish the communication between your workspace and the Semgrep Cloud Platform by following these steps:
+
+1. Create `SEMGREP_APP_TOKEN` by following the [Creating a `SEMGREP_APP_TOKEN`](/semgrep-ci/running-semgrep-ci-with-semgrep-app/#creating-a-semgrep_app_token) documentation. Copy the value of the token created in the Semgrep Cloud Platform.
+1. In the Bitbucket workspace, click the cogwheel icon next to your profile icon in the top right.
+1. Click **Workspace settings** > **Access tokens**.
+1. Click **Create Workspace Access Token**.
+1. Create a new workspace variable by entering **SEMGREP_APP_TOKEN** into the **Name** field, and then paste the value of the token in the **Value** field.
+1. Enable the **Secured** option, and then click **Add**.
```

Comment:
I think creating a SEMGREP_APP_TOKEN is needed in general to get Semgrep running in Bitbucket Cloud but we may already have these instructions elsewhere. I noticed in the sections for Github PR comments and Gitlab MR comments, the docs assumed that users already had Semgrep set up in their CI/CD so I was wondering if this would add more confusion to that flow. Not sure what's the convention here but will defer to you on it.

Reply:
@pangngan when we click add a new project in Projects page, one of the steps is to automatically generate a new App token. Do we recommend having separate tokens per repository? I can just interlink to token creation section. Note: It was definitely confusing for me. Although I already have Semgrep App token, I still failed to remember to copy it, so having it in one procedure and maybe just optionally skipping it felt more concise.

Reply:
@adamkvitek As I'm reading this on a second pass, I think there might be confusion on how to add the SEMGREP_APP_TOKEN to Bitbucket. 

I think what you're saying here is that we have to create a workspace access token for SEMGREP_APP_TOKEN, but that is not quite correct. All we have to do is add SEMGREP_APP_TOKEN as a workspace variable. 

A workspace access token is a personal access token created by Bitbucket so that we are able to ping the API. SEMGREP_APP_TOKEN is created by us so that the user can authenticate with us. What I'm trying to say is that all the user needs to do is add the SEMGREP_APP_TOKEN to Workspace variables. They do not have to create a workspace access token.

Reply:
Similarly, for those not on Bitbucket Cloud Premium, to add the SEMGREP_APP_TOKEN, they can just add it as a repository variable.

Reply:
Oh it was a typo! Thank you! https://github.com/returntocorp/semgrep-docs/pull/915/files#r1151561124

Reply:
@pangngan Sorry for resurrecting this but I agree that we should keep the SEMGREP_APP_TOKEN bits out of this. The idea for **PR comments** is to just create a PAT whether it's workspace or repository. 

@adamkvitek I suggest removing the SEMGREP_APP_TOKEN stuff.